src/backend/gitporter/gitprovider/github.py

```diff
@@ -8,17 +8,35 @@
 import httpx
 from .common import RemoteRepositoryInformation
 
+class GithubRepoItemOwner(t.TypedDict):
+    login: str
+    avatar_url: str
 
-class GithubRepoListResponse(t.TypedDict):
-    name: str
+class GithubRepoItem(t.TypedDict):
+    name: str 
     full_name: str
-    git_url: str
-    ssh_url: str
     clone_url: str
+    owner: GithubRepoItemOwner
 
-def get_remote_repositories(workspace: str) -> t.List[RemoteRepositoryInformation]:
-    response = httpx.get(f"https://api.github.com/users/{workspace}/repos")
+def get_remote_repositories(access_token: str) -> t.List[RemoteRepositoryInformation]:
+    github_headers = {
+        "Accept": "application/vnd.github+json", 
+        "Authorization": f"Bearer {access_token}", 
+        "X-GitHub-Api-Version": "2022-11-28",
+        }
+    
+    # TODO pagination. We can get a maximum of 100 repos at the moment
+    response = httpx.get("https://api.github.com/user/repos?affiliation=owner,organization_member&per_page=100", headers=github_headers)
     response.raise_for_status()
-    github_repositories: t.List[GithubRepoListResponse] = response.json()
+    github_repositories: t.List[GithubRepoItem] = response.json()
+    result = []
+    for github_item in github_repositories:
+        result.append(
+            RemoteRepositoryInformation(
+                repository_name=github_item["name"],
+                workspace_name=github_item["owner"]["login"],
+                clone_url=github_item["clone_url"].replace("https://", f"https://{access_token}@")
```

Comment:
was, wenn `http://`? (glaube `://` wäre besser als match)

Reply:
Ja, habe ich auch überlegt. Aber git liefert immer https als url, da es nur mit https gehen würde. Deswegen fände ich es gar nicht schlecht wenn wir durch einen "unauthorized access" in den error logs sehen, dass sich das geändert hat

Reply:
ok ok